.github/workflows/CI.yml

```diff
@@ -21,3 +25,31 @@ jobs:
     - name: Test
       run: yarn test
 
+  test_foundry:
+    strategy:
+      fail-fast: true
+
+    name: Foundry tests 
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Run Forge build
+        run: |
+          cd packages/contracts
```

Comment:
There's probably a better way to navigate here once and then preserve CWD across tasks, but it does the trick for now.